packages/eslint-plugin/docs/rules/strict-enums.md

```diff
@@ -0,0 +1,149 @@
+# `strict-enums`
+
+Disallows the usage of unsafe enum patterns.
+
+## Rule Details
+
+Horrifyingly, the TypeScript compiler will allow you to set any number to a variable containing a number enum, like this:
+
+```ts
+enum Fruit {
+  Apple,
+  Banana,
+}
+
+let fruit = Fruit.Apple;
+fruit = 999; // No error
+```
+
+This has resulted in many TypeScript programmers avoiding the use of enums altogether. Instead, they should use this rule, which bans working with enums in potentially unsafe ways.
+
+See the examples below for the types of patterns that are prevented.
+
+## Goals
+
+The goal of this rule is to make enums work like they do in other languages. One of the main benefits of enums is that they allow you to write code that is future-safe, because enums are supposed to be resilient to reorganization. If you arbitrarily change the values of an enum (or change the ordering of an enum with computed values), the idea is that nothing in your code-base should break.
+
+## Banned Patterns
+
+This rule bans:
+
+1. Enum incrementing/decrementing - `incorrectIncrement`
+1. Mismatched enum declarations/assignments - `mismatchedAssignment`
+1. Mismatched enum comparisons - `mismatchedComparison`
+1. Mismatched enum function arguments - `mismatchedFunctionArgument`
```

Comment:
I'd take https://github.com/typescript-eslint/typescript-eslint/pull/4864/files#r922968222 one step further and move the more full explanation from the bottom of the file up to here. That's good info down there - and users don't see message IDs.

```suggestion
```